jbmc/src/java_bytecode/lift_clinit_calls.cpp

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************\
+
+Module: Static initializer call lifting
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+/// file Static initializer call lifting
+
+#include "lift_clinit_calls.h"
+
+#include <java_bytecode/java_static_initializers.h>
+
+#include <util/expr_iterator.h>
+
+#include <algorithm>
+
+codet lift_clinit_calls(codet input)
+{
+  // 1. Gather any clinit calls present in `input`:
+  std::vector<symbol_exprt> clinit_wrappers_called;
+
+  for(auto it = input.depth_begin(), itend = input.depth_end(); it != itend;
+      ++it)
+  {
+    if(const auto code = expr_try_dynamic_cast<codet>(*it))
+    {
+      if(code->get_statement() == ID_function_call)
+      {
+        if(
+          const auto callee = expr_try_dynamic_cast<symbol_exprt>(
+            to_code_function_call(*code).function()))
+        {
+          if(is_clinit_wrapper_function(callee->get_identifier()))
+          {
+            clinit_wrappers_called.push_back(*callee);
+            // Replace call with skip:
+            it.mutate() = code_skipt();
+          }
+        }
+      }
+    }
+  }
+
+  if(clinit_wrappers_called.empty())
+    return input;
```

Comment:
Since you're explicitly moving all other returns, perhaps:
```
  return std::move(input);
```

Reply:
I actually mustn't -- when I return a `codet` I must not `move` so that RVO works (otherwise Clang warns, and we have `-Werror`), whereas if I return a type that would require a conversion (e.g. `code_blockt`) then I must use `std::move`.